tests/test_operation.py

```diff
@@ -908,6 +908,70 @@ def test_prune(self, tensor_observable, expected):
         assert type(O_pruned) == type(expected)
         assert O_pruned.wires == expected.wires
 
+    def test_prune_while_queueing_return_tensor(self):
```

Comment:
I don't quite follow, what is the difference between these two new tests?

Reply:
Oh, `qml.operation.Tensor(a, c).prune()` results in an observable (the one that is left when removing the identity), while `qml.operation.Tensor(a, b, c).prune()` remains a tensor (of the two observables left). Is there any way I can explain better?

Reply:
Stricyly speaking the second test is not so urgent, but I had it and thought it can't harm to add

Reply:
Ohhhh okay I think I see - theoretically they are both observables, but in the second case when where's more than one and they are tensored together then they are also type `Tensor`?